test/compilable/test5227.d

```diff
@@ -0,0 +1,162 @@
+/*
+REQUIRED_ARGS:
+PERMUTE_ARGS:
+TEST_OUTPUT:
+---
+log()
+1.70475L
+1.70475L
+1.70475L
+log2()
+2.45943L
+2.45943L
+2.45943L
+log10()
+0.740363L
+0.740363L
+0.740363L
+nearbyint()
+6.00000L
+6.00000L
+6.00000L
+rint()
+6.00000F
+6.00000
+6.00000L
+round()
+6.00000L
+6.00000L
+6.00000L
+floor()
+5.00000F
+5.00000
+5.00000L
+ceil()
+6.00000F
+6.00000
+6.00000L
+trunc()
+5.00000L
+5.00000L
+5.00000L
+expm1()
+243.692L
+243.692L
+243.692L
+exp2()
+45.2548L
+45.2548L
+45.2548L
+fmin()
+-3.2L
+-3.2L
+-3.2L
+fmax()
+5.2L
+5.2L
+5.2L
+copysign()
+-2.5F
+-2.5
+-2.5L
+pow()
+9.88212F
+9.88212
+9.88212L
+9.88212
+fma()
+-12.84L
+-12.84L
+-12.84L
+---
+*/
+
+// https://issues.dlang.org/show_bug.cgi?id=5227
+
+import std.math;
+
+pragma(msg, "log()");
+enum logf = log(5.5f); pragma(msg, logf);
```

Comment:
How come `pragma(msg)` is used to test this instead of `static assert`?

Reply:
Right. The current tests also test that CTFE `log(float|double)` etc. return a `real`, i.e., that there are no float/double overloads, which I'm not sure is desirable.

Reply:
The `std.math.log` function does not have overloads, and passing float/double to it will return a real. I don't think CTFE should do something different than the corresponding std.math function.

Reply:
> I don't think CTFE should do something different than the corresponding std.math function.

That isn't the question, it's whether the float/double tests should be there if there's no corresponding function. As soon as Phobos gets proper overloads, these DMD tests will fail, so DMD has to be patched in tandem. I'm in favor of adding corresponding CTFE tests after extending Phobos.

Reply:
Put the functions in Phobos first, then in dmd. Not the other way around, because they cannot be tested if put in dmd first.

Reply:
I'd also recommend skipping these tests for now: Testing them here to return real will block adding the overloads to phobos.

Reply:
> Put the functions in Phobos first, then in dmd. Not the other way around, because they cannot be tested if put in dmd first.

Absolutely, but this patch here puts them in dmd first, that's what I've been trying to tell the whole time.